### PR TITLE
pkg/asset/manifests/infrastructure: Set status.platformStatus.aws.region

### PR DESCRIPTION
**Pull request by wking**
Builds on #1718; I'll rebase after that lands.

Using openshift/api@dedfb47b1f (openshift/api#300) so the registry can use it as a default for their [config][1] [resource][2].

This will allow the registry to move off the deprecated cluster-config-v1 (#680) and make it easier for them to move from `openshiftClusterID` tagging to `kubernetes.io/cluster/{infraName}` tagging (openshift/cluster-image-registry-operator#267), which is part of addressing [rhbz#1685089][3].

[1]: https://github.com/openshift/cluster-image-registry-operator/blob/670539cda460d97e0a3e23f077fdf9dd00d2ccdc/manifests/00-crd.yaml
[2]: https://github.com/openshift/cluster-image-registry-operator/blob/670539cda460d97e0a3e23f077fdf9dd00d2ccdc/pkg/apis/imageregistry/v1/types.go#L179-L184
[3]: https://bugzilla.redhat.com/show_bug.cgi?id=1685089#c11

### CODE REVIEW

File: pkg/asset/manifests/infrastructure.go [hunk 1]
```diff
@@ -81,14 +81,24 @@ func (i *Infrastructure) Generate(dependencies asset.Parents) error {
...
+			PlatformStatus: &configv1.PlatformStatus{
+				Type: platform,
+			},
 			APIServerURL:         getInternalAPIServerURL(installConfig.Config),
 			APIServerInternalURL: getInternalAPIServerURL(installConfig.Config),
 			EtcdDiscoveryDomain:  getEtcdDiscoveryDomain(installConfig.Config),
 		},
 	}
 
+	switch installConfig.Config.Platform.Name() {
```
Comment (abhinavdahiya) [inline pkg/asset/manifests/infrastructure.go]: could we collapse the two switches for platforms in to one?

Comment (wking) [reply pkg/asset/manifests/infrastructure.go]: > could we collapse the two switches for platforms in to one?

Done with c52831893 -> ecaf8bfbf.

**Comment by wking** [comment]
Rebased onto master with ecaf8bfbf -> 8b3b37571 now that #1718 has landed.  Also fixed [the `nil`-dereference bug][1].

[1]: https://storage.googleapis.com/origin-ci-test/pr-logs/pull/openshift_installer/1725/pull-ci-openshift-installer-master-e2e-aws/5710/build-log.txt

**Comment by abhinavdahiya** [comment]
@wking We would need a plan to set this up in existing clusters to make upgrades possible.

 Define a job _in some operartor_ that ends up in release image that sets this field from cluster-config-v1.
before we merge this change

File: pkg/asset/manifests/infrastructure.go [hunk 2]
```diff
@@ -82,13 +64,34 @@ func (i *Infrastructure) Generate(dependencies asset.Parents) error {
...
+	case none.Name:
+		config.Status.PlatformStatus.Type = configv1.NonePlatformType
+	case libvirt.Name:
+		config.Status.PlatformStatus.Type = configv1.LibvirtPlatformType
+	case openstack.Name:
+		config.Status.PlatformStatus.Type = configv1.OpenStackPlatformType
+	case vsphere.Name:
+		config.Status.PlatformStatus.Type = configv1.VSpherePlatformType
+	case azure.Name:
+		config.Status.PlatformStatus.Type = configv1.AzurePlatformType
```
Comment (dmage) [inline pkg/asset/manifests/infrastructure.go]: We need the same for Azure.

**Comment by coreydaley** [comment]
@wking bump

**Comment by sgreene570** [comment]
friendly ping on this :smiley:

**Comment by celebdor** [comment]
@abhinavdahiya @wking any idea when this will get in? @cybertron needs to add info to the Baremetal PlatformStatus and I'm wondering if this needs to be getting merged first.

**Pull request closed without merging**